diff -Naur smtpclient-1.0.0/smtpclient.1 smtpclient-1.0.0-p/smtpclient.1
--- smtpclient-1.0.0/smtpclient.1	Tue Oct 21 18:53:45 1997
+++ smtpclient-1.0.0-p/smtpclient.1	Tue Oct 12 15:55:11 2004
@@ -201,6 +201,7 @@
 [\fB\-M\fR]
 [\fB\-L\fR]
 [\fB\-v\fR]
+[\fB\-R\fR]
 \fIrecipient\fR ...
 .PP
 \fBsmtpclient\fR 
@@ -246,6 +247,8 @@
 Use \s-1MIME\s0\-style translation to quoted-printable (base 16).
 .Ip "\fB\-L\fR, \fB--use-syslog\fR" 4
 Log errors to system's \fIsyslog\fR facility instead of \fIstderr\fR.
+.Ip "\fB\-R\fR, \fB--raw\fR" 4
+Send raw message, without adding any headers.
 .Ip "\fB\-v\fR, \fB--verbose\fR" 4
 Turn on verbose logging to \fIstdout\fR.
 .Ip "\fB\-V\fR, \fB--version\fR" 4
diff -Naur smtpclient-1.0.0/smtpclient_main.c smtpclient-1.0.0-p/smtpclient_main.c
--- smtpclient-1.0.0/smtpclient_main.c	Tue Oct 21 18:56:27 1997
+++ smtpclient-1.0.0-p/smtpclient_main.c	Tue Oct 12 15:59:47 2004
@@ -65,6 +65,7 @@
 static int   mime_style = 0;
 static int   verbose    = 0;
 static int   usesyslog  = 0;
+static int   raw_message = 0;
 
 static FILE *sfp;
 static FILE *rfp;
@@ -108,6 +109,7 @@
     fprintf(stderr, "  -r, --reply-to=ADDR    address of the sender for replies\n");
     fprintf(stderr, "  -e, --errors-to=ADDR   address to send delivery errors to\n");
     fprintf(stderr, "  -c, --carbon-copy=ADDR address to send copy of message to\n");
+    fprintf(stderr, "  -R, --raw              send raw message, without processing headers\n");
     fprintf(stderr, "\n");
     fprintf(stderr, "Processing Options:\n");
     fprintf(stderr, "  -S, --smtp-host=HOST   host where MTA can be contacted via SMTP\n");
@@ -292,7 +294,8 @@
     { "use-syslog",   0, NULL, 'L' },
     { "verbose",      0, NULL, 'v' },
     { "version",      0, NULL, 'V' },
-    { "help",         0, NULL, 'h' }
+    { "help",         0, NULL, 'h' },
+    { "raw",	      0, NULL, 'R' }
 };
 
 int main(int argc, char **argv)
@@ -317,7 +320,7 @@
     /*
      *  Parse options
      */
-    while ((c = getopt_long(argc, argv, ":s:f:r:e:c:S:P:MLvVh", options, NULL)) != EOF) {
+    while ((c = getopt_long(argc, argv, ":s:f:r:e:c:S:P:MLRvVh", options, NULL)) != EOF) {
         switch (c) {
             case 's':
                 subject = optarg;
@@ -355,6 +358,9 @@
             case 'h':
                 usage();
                 exit(0);
+	    case 'R':
+		raw_message = 1;
+		break;
             default:
                 fprintf(stderr, "SMTP: invalid option `%c'\n", optopt);
                 fprintf(stderr, "Try `%s --help' for more information.\n", argv[0]);
@@ -457,6 +463,7 @@
         chat("RCPT TO: <%s>\r\n", cc_addr);
     chat("DATA\r\n");
 
+    if (!raw_message) {
     /* 
      *  Give out Message header. 
      */
@@ -488,11 +495,12 @@
     }
 
     fprintf(sfp, "\r\n");
+    }
 
     /* 
      *  Give out Message body.
      */
-    if (mime_style) {
+    if (mime_style && (!raw_message)) {
         toqp(stdin, sfp);
     } else {
         while (fgets(buf, sizeof(buf), stdin)) {
